src/python/pants/help/help_printer.py

```diff
@@ -78,14 +70,90 @@ def print_hint() -> None:
             return 1
         return 0
 
+    def _print_alternatives(self, match: str, all_things: Iterable[str]) -> None:
+        did_you_mean = list(difflib.get_close_matches(match, all_things))
+
+        if did_you_mean:
+            formatted_matches = self._format_did_you_mean_matches(did_you_mean)
+            print(f"Did you mean {formatted_matches}?")
+
     def _print_title(self, title_text: str) -> None:
         title = self.maybe_green(f"{title_text}\n{'-' * len(title_text)}")
         print(f"\n{title}\n")
 
+    def _print_table(self, table: Dict[str, Optional[str]], indent: int = 0) -> None:
+        longest_key = max(len(key) for key, value in table.items() if value is not None)
+        for key, value in table.items():
+            if value is None:
+                continue
+            print(
+                self.maybe_cyan(f"{key:{longest_key}}:"),
+                self.maybe_magenta(
+                    f"\n{' ':{longest_key+2}}".join(
+                        hard_wrap(value, width=self._width - longest_key - 2)
+                    )
+                ),
+            )
+
+    def _get_thing_help_table(self) -> Dict[str, Callable[[str, bool], None]]:
+        def _help_table(
+            things: Iterable[str], help_printer: Callable[[str, bool], None]
+        ) -> Dict[str, Callable[[str, bool], None]]:
+            return dict(zip(things, cycle((help_printer,))))
+
+        top_level_help_items = _help_table(self._reserved_names, self._print_top_level_help)
+        return {
+            **top_level_help_items,
+            **_help_table(self._all_help_info.scope_to_help_info.keys(), self._print_options_help),
+            **_help_table(
+                self._all_help_info.name_to_target_type_info.keys(), self._print_target_help
+            ),
+            **_help_table(
+                self._all_help_info.rule_output_type_to_rule_infos.keys(), self._print_api_type_help
+            ),
+        }
+
+    def _disambiguate_things(
+        self, things: Iterable[str], all_things: Iterable[str]
+    ) -> Tuple[Set[str], Set[str]]:
+        """Returns two sets of strings, one with disambiguated things and the second with
+        unresolvable things."""
```

Comment:
Callers should maybe render a warning that something wasn't a perfect match...?

Reply:
There's just one caller, and it does that:

https://github.com/pantsbuild/pants/pull/15646/commits/98b487a68c8aa9173861847543d90440f7aff106#diff-cb9b4becc071bc0af2861cfa32a987ee88c71f9b046ac794a973499044e5ebdaR167-R174